File: atcoder/371/B.py
def main():
    N, M = map(int, input().split())
    hastaro = [False for _ in range(N)]
    for i in range(M):
        a, b = input().split()
        a = int(a) - 1
        if hastaro[a] == False and b =='M':
            print("Yes")
            hastaro[a] = True
        else:
            print("No")
            
if __name__ == "__main__":
    main()

File: atcoder/371/B.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int main()
{
    int N, M;
    cin >> N >> M;
    vector<bool> hastaro(N + 1, false);
    int i;
    for (i = 0; i < M; i++)
    {
        int a;
        char b;
        cin >> a >> b;
        if (hastaro[a] == false && b == 'M')
        {
            cout << "Yes" << endl;
            hastaro[a] = true;
        }
        else
        {
            cout << "No" << endl;
        }
    }
    return 0;
}